api/v1alpha1/datadogagent_types.go

```diff
@@ -375,6 +377,20 @@ type ProcessSpec struct {
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
+// KubeStateMetricsCore contains the required parameters to enable and override the configuration
+// of the Kubernetes State Metrics Core (aka v2.0.0) of the check.
+// +k8s:openapi-gen=true
+type KubeStateMetricsCore struct {
+	// Enable this to start the Kubernetes State Metrics Core check.
+	// Refer to https://github.com/DataDog/datadog-operator/blob/master/docs/kubernetes_state_metrics.md
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// To override the configuration for the default Kubernetes State Metrics Core check.
+	// Must point to a ConfigMap containing a valid cluster check configuration.
+	Conf *string `json:"conf,omitempty"`
```

Comment:
to configure the configmap, I think we can reuse some go struct that we already use in this project like in [CustomConfigSpec](https://github.com/DataDog/datadog-operator/blob/cba7872e332dfbf2db8d2e2397bd20d4e25815f9/api/v1alpha1/datadogagent_types.go#L561) we have the [ConfigFileConfigMapSpec](https://github.com/DataDog/datadog-operator/blob/cba7872e332dfbf2db8d2e2397bd20d4e25815f9/api/v1alpha1/datadogagent_types.go#L550) use to provide the ConfigMap information.

Maybe we need to rename the `ConfigFileConfigMapSpec` in a more generic way.